lib/compilers/hlsl.ts

```diff
@@ -24,28 +24,47 @@
 
 import path from 'path';
 
+import {ParseFilters} from '../../types/features/filters.interfaces';
 import {BaseCompiler} from '../base-compiler';
 
 export class HLSLCompiler extends BaseCompiler {
     static get key() {
         return 'hlsl';
     }
 
-    constructor(info, env) {
+    constructor(info: any, env: any) {
         super(info, env);
 
         this.compiler.supportsIntel = false;
     }
 
     /* eslint-disable no-unused-vars */
-    optionsForFilter(filters, outputFilename) {
+    override optionsForFilter(filters: ParseFilters, outputFilename: any, userOptions?: any): any[] {
```

Comment:
The proper types for this are missing in the base class, sorry! I've fixed it in #3966
But for now you can set them to:

```suggestion
    override optionsForFilter(filters: ParseFilters, outputFilename: string, userOptions?: string[]): string[] {
```